cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/Relation.java

```diff
@@ -104,4 +107,24 @@ default Relation window(WindowAggregationDefinition aggregationDefinition) {
   default Relation deduplicate(DeduplicateAggregationDefinition aggregationDefinition) {
     return new InvalidRelation("Deduplicate is unsupported");
   }
+
+  /**
+   * Get the {@link Schema} of the relation if available, null otherwise.
+   *
+   * @return the schema of the relation.
+   */
+  @Nullable
+  default Schema getSchema() {
+    return null;
+  }
+
+  /**
+   * Adds schema information to the relation. This can be used for validation purposes.
+   *
+   * @param schema The schema.
+   * @return A new relation with the schema added.
+   */
+  default Relation addSchema(@Nullable Schema schema) {
```

Comment:
This should not be here. Getting a schema is ok, but not applying a schema

Reply:
Who is "we"? I mean, the means to do it highly depends on if that's platform, calcite or engine. Generally, schema should be specified during Relation object creation. But this is not something that should be available through interfaces. It should only be possible as part of Relation creation logic.

It's much better to introduce new methods along with it's usage. It should answer such questions.